pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -276,6 +280,13 @@ def metrics(self) -> Dict[MetricSource, Dict[str, _METRIC]]:
         on_step = not self._epoch_end_reached
         return self.trainer._results.metrics(on_step)
 
+    @property
+    def gpus_metrics(self) -> Dict[str, str]:
+        if self.trainer._device_type == DeviceType.GPU and self.log_gpu_memory:
+            mem_map = memory.get_memory_profile(self.log_gpu_memory)
+            self._gpus_metrics.update(mem_map)
+        return self._gpus_metrics
+
```

Comment:
Why did this PR need to add `_gpu_metrics`? Doesn't seem related to the issue linked.

This means that gpu metrics are now duplicated in this dictionary and in logged metrics.

Also it only gets filled when `self.log_gpu_memory` so it can't be used anyways without the flag.

Reply:
I think this broke `log_gpu_memory="min_max"` option in `Trainer.fit`. Looks related to: https://github.com/PyTorchLightning/pytorch-lightning/issues/9010

Essentially `memory. get_memory_profile` adds two keys that are not in conventional format `min_gpu_mem` and `max_gpu_mem` (typically keys are in `f"gpu_id: {gpu_id}/memory.used (MB)"`